### PR TITLE
App: Final PR

### PR DESCRIPTION
**Pull request by ivaan1992**
This is my final PR.
For this project I:

- Create a new react app.
- Add components.
- Install linters.
- Add a redux store. 
- Add routes.
- Test the components. 
- Display my project in netlify.
- Record a <a href="https://www.loom.com/share/362b3687ca0f45ff94b3eede4426cae5">loom video</a>.

### CODE REVIEW

File: src/components/Episodes/EpisodeCard.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            <h3 className="title"></h3>
+            <p className="airdate"></p>
+            <div className="character-container">
+
+            </div>
+        </div>
+    )
+}
+
+export default EpisodeCard;  */
```
Comment (hiromataba) [inline src/components/Episodes/EpisodeCard.jsx:20]: - [ ] Please remove commented code to have cleaner code and readable

File: src/redux/Api/base.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/* const BASE_URL = 'https://rickandmortyapi.com/api';
+
+const connectApi = async (page = 1, type) => {
+  const response = await fetch(`${BASE_URL}/${type}?page=${page}`);
+  const data = await response.json();
+  return data;
+};
+
+export default connectApi;  */
```
Comment (hiromataba) [inline src/redux/Api/base.js:9]: - [ ] Please remove the commented code to make your code cleaner and more readable

File: src/redux/Characters/CharactersRedux.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    dispatch(characters(name, NAME));
+    dispatch(characters(results, CHARACTERS));
+  }).catch(() => {});
+};
+
+/* export const getName = (name, dispatch) => {
+  connectCharApi(name).then(({ search }) => {
+    dispatch(characters(search));
+  });
+}; */
```
Comment (hiromataba) [inline src/redux/Characters/CharactersRedux.js:27]: - [ ] Please remove the commented code to make your code cleaner and more readable

File: src/components/Search/SearchBar.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  };
+
+  //   const filter = (searchByName) => {
+  //     const searchResult = getCharacter.filter((char) => {
+  //       if (char.name.toString().toLowerCase().includes(searchByName.toLowerCase)) {
+  //         return char;
+  //       }
+  //     });
+  //   };
```
Comment (hiromataba) [inline src/components/Search/SearchBar.js:37]: - [ ] Please remove the commented code to make your code cleaner and more readable

File: src/components/Search/SearchBar.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    const nameState = useSelector((state) => state.characterReducer);
+    const dispatch = useDispatch();
+    const resultSearch = nameState.filter((element) => {
+      if (element.name.toString().toLowerCase().include(searchName.toLowerCase())) {
+        return element;
+      } return element;
+    });
+
+    dispatch(resultSearch);
+  }; */
```
Comment (hiromataba) [inline src/components/Search/SearchBar.js:18]: - [ ] Please remove the commented code to make your code cleaner and more readable

File: src/components/Location/LocationInfo.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    <img src={residents} alt={name} />
+  </div>
+);
+
+LocationInfo.propTypes = {
+  name: PropTypes.string.isRequired,
+  type: PropTypes.string.isRequired,
+  dimension: PropTypes.string.isRequired,
+  residents: PropTypes.string.isRequired,
+  // image: PropTypes.string.isRequired,
```
Comment (hiromataba) [inline src/components/Location/LocationInfo.js:20]: - [ ] Please remove the commented code to make your code cleaner and more readable

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @ivaan1992      :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Nice commit history :heavy_check_mark: 
- Profesional README file :rocket: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: .babelrc [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (sheylaPozo) [inline .babelrc:1]: ## Status: ✔️ Project Approved ✔️ 
----

Hello @ivaan1992 🔥 

> # Excellent work on your project and completing the requested changes, it looks very professional and it's completed. Congrats on the good work!

surprised

Now you can celebrate! 🥇 🚀 👍 💯 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️
`There's nothing left to say but time to merge! 😊`

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sheylaPozo — APPROVED** [review]

**Comment by sheylaPozo** [comment]
- [x] Your project looks really good and complete, good job in following the previous reviewer's suggestions.✨✔

**Pull request merged**